driver-core/src/main/com/mongodb/operation/ChangeStreamBatchCursor.java

```diff
@@ -0,0 +1,164 @@
+import static java.util.Collections.singletonList;
+
+final class ChangeStreamBatchCursor<T> implements BatchCursor<T> {
+    private static final BsonDocumentCodec BSON_DOCUMENT_CODEC = new BsonDocumentCodec();
+    private final ReadBinding binding;
+    private final ChangeStreamOperation<T> changeStreamOperation;
+
+    private volatile BsonDocument resumeToken;
+    private volatile QueryBatchCursor<RawBsonDocument> wrapped;
+
+    ChangeStreamBatchCursor(final ChangeStreamOperation<T> changeStreamOperation, final ReadBinding binding,
+                            final QueryBatchCursor<RawBsonDocument> wrapped) {
+        this.changeStreamOperation = changeStreamOperation;
+        this.resumeToken = changeStreamOperation.getResumeToken();
+        this.binding = binding;
+        this.wrapped = wrapped;
+    }
+
+    QueryBatchCursor<RawBsonDocument> getWrapped() {
+        return wrapped;
+    }
+
+    @Override
+    public boolean hasNext() {
+        return resumeAbleOperation(new Function<QueryBatchCursor<RawBsonDocument>, Boolean>() {
+            @Override
+            public Boolean apply(final QueryBatchCursor<RawBsonDocument> queryBatchCursor) {
+                return queryBatchCursor.hasNext();
+            }
+        });
+    }
+
+    @Override
+    public List<T> next() {
+        return resumeAbleOperation(new Function<QueryBatchCursor<RawBsonDocument>, List<T>>() {
+            @Override
+            public List<T> apply(final QueryBatchCursor<RawBsonDocument> queryBatchCursor) {
+                return convertResults(queryBatchCursor.next());
+            }
+        });
+    }
+
+    @Override
+    public List<T> tryNext() {
+        return resumeAbleOperation(new Function<QueryBatchCursor<RawBsonDocument>, List<T>>() {
+            @Override
+            public List<T> apply(final QueryBatchCursor<RawBsonDocument> queryBatchCursor) {
+                return convertResults(queryBatchCursor.tryNext());
+            }
+        });
+    }
+
+    @Override
+    public void close() {
+        wrapped.close();
+    }
+
+    @Override
+    public void setBatchSize(final int batchSize) {
+        wrapped.setBatchSize(batchSize);
+    }
+
+    @Override
+    public int getBatchSize() {
+        return wrapped.getBatchSize();
+    }
+
+    @Override
+    public ServerCursor getServerCursor() {
+        return wrapped.getServerCursor();
+    }
+
+    @Override
+    public ServerAddress getServerAddress() {
+        return wrapped.getServerAddress();
+    }
+
+    @Override
+    public void remove() {
+        throw new UnsupportedOperationException("Not implemented yet!");
+    }
+
+    private List<T> convertResults(final List<RawBsonDocument> rawDocuments) {
+        List<T> results = null;
+        if (rawDocuments != null) {
+            results = new ArrayList<T>();
+            for (RawBsonDocument rawDocument : rawDocuments) {
+                resumeToken = rawDocument.getDocument("_id");
```

Comment:
The spec says "A driver MUST raise an error on the first response received without a resume token".  I read that as saying that this should throw a MongoChangeStreamException if _id is missing or is not a document.  As is, it will throw a BsonInvalidOperationException. 

And doing that here means you can get rid of the check for a missing resumeToken that'
s now in ChangeStreamOperation.